tempesta_fw/ss_skb.c

```diff
@@ -1036,12 +1042,18 @@ __copy_ip_header(struct sk_buff *to, const struct sk_buff *from)
 	const struct iphdr *ip4 = ip_hdr(from);
 	const struct ipv6hdr *ip6 = ipv6_hdr(from);
 
+	/*
+	 * Place IP header just after link layer headers,
+	 * see definitions of MAX_TCP_HEADER and MAX_IP_HDR_LEN.
+	 * Note that only new skbs allocated by ss_skb_alloc() are used here,
+	 * so all of them have reserved MAX_TCP_HEADER areas.
+	 */
+	BUG_ON(skb_headroom(to) < MAX_TCP_HEADER);
+	skb_set_network_header(to, -128);
```

Comment:
Perhaps, there's another way of doing this other than doing a subtraction of *_magic_* number. Like, `MAX_TCP_HEADER - MAX_HEADER`?